contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/example_otio_reader_plugin.py

```diff
@@ -0,0 +1,168 @@
+                   ("source-group-complete",
+                    self.source_group_complete,
+                    "Expand otio supported files (synchronous load)"),
+                   ("after-progressive-loading",
+                    self.after_progressive_loading,
+                    "Expand otio supported files (asynchronous load)")],
+                  None)
+
+        # Start as sleeping
+        self.mode = Mode.sleeping
+
+    def incoming_source_path(self, event):
+        """
+        Detects if otio file is being loaded, and replaces with
+        empty movie proc containing an otioFile tag. This will be
+        replaced in after loading.
+        """
+        event.reject()
+
+        parts = event.contents().split(';')
+        in_path = parts[0]
+        _, ext = os.path.splitext(in_path)
+        if ext:
+            ext = ext[1:]
+
+        if ext in otio.adapters.suffixes_with_defined_adapters(read=True):
+            self.mode = Mode.loading
+            movieproc = 'blank,otioFile={}.movieproc'.format(in_path)
+            event.setReturnContent(movieproc)
+
+    def after_progressive_loading(self, event):
+        """
+        After progress loading event, is used for asynchronous addSource
+        file loading.
+        """
+        event.reject()
+
+        if self.mode != Mode.loading:
+            return
+        self.mode = Mode.processing
+        self.expand_sources()
+
+    def source_group_complete(self, event):
+        """
+        Source group complete event, is used for synchronous addSource
+        file loading.
+        """
+        event.reject()
+        if self.mode == Mode.sleeping:
+            # this plugin isn't doing anything
+            return
+
+        if self.mode == Mode.processing:
+            # already processing otio
+            return
+
+        if commands.loadTotal() > 0:
+            # async load
+            return
+
+        self.mode = Mode.processing
+        self.expand_sources()
+
+    def expand_sources(self):
+        """
+        Expand any movie movieproc otioFile sources.
+        """
+        # disable caching for load speed
+        cache_mode = commands.cacheMode()
+        commands.setCacheMode(commands.CacheOff)
+
+        try:
+            # find sources with a movieproc with an otioFile=foo.otio tag
+            default_inputs, _ = commands.nodeConnections('defaultSequence')
+            for src in commands.nodesOfType('RVSource'):
+                src_group = commands.nodeGroup(src)
+                if src_group not in default_inputs:
+                    # not in default sequence, already processed
+                    continue
+
+                # get the source file name
+                paths = [info.get('file', '') for info in
+                         commands.sourceMediaInfoList(src)]
+                for info_path in paths:
+                    # Looking for: 'blank,otioFile=/foo.otio.movieproc'
+                    parts = info_path.split("=", 1)
```

Comment:
In line 132 there is a default value of `''` used for paths.  It won't throw an exception here (since line 138 will throw it out), but I think for clarity you could add a:
```python
if not info_path:
  continue
```
to skip those cases altogether.

Reply:
Good catch! I'll fix that.